service-loadbalancer/gce/backends.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+
+	compute "google.golang.org/api/compute/v1"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/golang/glog"
+)
+
+type Backends struct {
+	cloud          BackendServices
+	instanceGroups InstanceGroups
+	pool           *poolStore
+	defaultIg      *compute.InstanceGroup
+	defaultBackend *compute.BackendService
+	defaultHc      *compute.HttpHealthCheck
+}
+
+func portKey(port int64) string {
+	return fmt.Sprintf("%v", port)
+}
+
+func beName(port int64) string {
+	return fmt.Sprintf("%v-%v", backendPrefix, port)
+}
+
+func NewBackendPool(
+	cloud BackendServices,
+	defaultBackendNodePort int64,
+	defaultIg *compute.InstanceGroup,
+	defaultHc *compute.HttpHealthCheck,
+	instanceGroups InstanceGroups) (BackendPool, error) {
+	backends := &Backends{
+		cloud:          cloud,
+		instanceGroups: instanceGroups,
+		pool:           newPoolStore(),
+		defaultIg:      defaultIg,
+		defaultHc:      defaultHc,
+	}
+	err := backends.Add(defaultBackendNodePort)
+	if err != nil {
+		return nil, err
+	}
+	backends.defaultBackend, err = backends.Get(defaultBackendNodePort)
```

Comment:
Hmm, Get just gets the backend, I want to assign to the defaultBackend member before returning the object from the constructor